35-Search_Insert_Position.py

```diff
@@ -0,0 +1,60 @@
+# -*- coding: utf-8 -*-
+'''
+Given a sorted array and a target value, return the index if the target is found.
+If not, return the index where it would be if it were inserted in order.
+
+You may assume no duplicates in the array.
+
+Example 1:
+Input: [1,3,5,6], 5
+Output: 2
+
+Example 2:
+Input: [1,3,5,6], 2
+Output: 1
+
+Example 3:
+Input: [1,3,5,6], 7
+Output: 4
+
+Example 4:
+Input: [1,3,5,6], 0
+Output: 0
+'''
+
+class Solution:
+    # Time complexity: O(n)
+    # Space complexity:  O(1)
+    def searchInsert(self, nums, target):
+        """
+        :type nums: List[int]
+        :type target: int
+        :rtype: int
+        """
+        for i in range(len(nums)):
+            if nums[i] >= target:
+                return i
+        return len(nums)
+
+class Solution2:
+    # Time complexity: O(1)
+    # Space complexity: O(1)
+    def searchInsert(self, nums, target):
+        """
+        :type nums: List[int]
+        :type target: int
+        :rtype: int
+        """
+        nums.append(target)
+        return sorted(nums).index(target)
+
+class Solution3:
+    # Time complexity: O(n)
```

Comment:
Since this solution has no advantage over solution1 in any case (time, space at (best, worst, average)), you may remove it.